grove/measurements/estimation.py

```diff
@@ -0,0 +1,161 @@
+"""
+Utilities for estimating expected values of Pauli terms given pyquil programs
+"""
+import numpy as np
+from pyquil.paulis import PauliSum, PauliTerm, commuting_sets, sI
+from pyquil.quil import Program
+from pyquil.gates import RY, RX
+
+
+class CommutationError(Exception):
+    pass
+
+
+def remove_imaginary_terms(pauli_sums):
+    """
+    Remove the imaginary components of the Pauli sum term
+
+    :param pauli_sums:
+    :return:
+    """
+    if not isinstance(pauli_sums, PauliSum):
+        raise TypeError("not a pauli sum. please give me one")
+    new_term = sI(0) * 0.0
+    for term in pauli_sums:
+        if (not np.isclose(term.coefficient.real, 0.0) and
+                np.isclose(term.coefficient.imag, 0.0)):
+            new_term += term
```

Comment:
So what if I pass a PauliSum that is ```sI(0) * (1.0 + 1j) + sX(0)```. This would imply that the first term is removed, but shouldn't its hermitian component survive?

Reply:
I agree.  I will update the method.